Prevent the logger from throwing a DisallowedHost error

abloom opened the pull request:
This PR includes 2 changes, one in each commit:

1. Switch from `syslog.severity` to `level` so that DD automatically pics up the log level
2. I found that my app was trying to log a `DisallowedHost` error when the logger would raise another one. This change prevents the logger from throwing a `DisallowedHost` error.

abloom commented:
@lnagel I'm not sure what the contribution policy is for this repo but based on the history it seems like you might be the person with the keys. If not, would you mind helping me find the right person to see about getting this merged?

lnagel commented:
The contribution policy is that contributions are welcome.

1. The switch away from `syslog.severity` seems unusual for me. What problem are you trying to solve? We are following the guidelines set here: https://docs.datadoghq.com/logs/log_configuration/attributes_naming_convention/#syslog-and-log-shippers

2. The DisallowedHost issue I can understand, this should be merged. However you can work around temporarly by restricting the allowed hostnames in your proxy config (e.g. in nginx)

abloom commented:
@lnagel thank you!

1. You're right, my `syslog.severity` change was unorthodox and I'm going to roll it back. It took me a while to respond to your feedback because I had to learn a bit more about datadog to understand what was going on. Long story short, I'm using Heroku and at some point an integration added a logging pipeline with a number of very helpful steps, but it also includes a "status remapper that uses `level`, not `syslog.severity`. Now that I better understand how this all works I've fixed this in our account and removed that commit.
2. Since we're using Heroku we have a pretty limited DNS surface area so it's pretty safe, but there are a few ways to trigger it and I happened to hit them while testing.

I think this is ready for a re-review

lnagel commented:
Released in v0.5.2
